src/components/FlatTreeRenderer.js

```diff
@@ -0,0 +1,436 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import _ from 'lodash'
+import { connect } from 'react-redux'
+import { animated, useTransition } from 'react-spring'
+import useMeasure from '../hooks/useMeasure.js'
+import ContentEditable from 'react-contenteditable'
+import { store } from '../store.js'
+
+// util
+import { contextOf, isDescendant, pathToContext, treeToFlatArray } from '../util'
+
+/** Wait for x ms. */
+// eslint-disable-next-line
+const wait = time => new Promise(
+  res => setTimeout(() => res(), time)
+)
+
+const DISTANT_THOUGHT_OPACITY = 0.5
+const TEXT_SELECTION_OPCAITY = 0.3
+
+// React spring config
+// const SPRING_CONFIG = { mass: 1, tension: 200, friction: 30, clamp: true }
+const SPRING_CONFIG_GROUP = { clamp: true }
+
+// factor by which second column of table view slides to right
+const TABLE_SECOND_COLUMN_OFFSET = 0
+
+// factor by which first column of table view slides to right
+const TABLE_FIRST_COLUMN_OFFSET = 1.5
+
+const TABLE_FIRST_COLUMN_WIDTH = 8
+
+// factor by which x offset should be increase with each depth
+const DEPTH_OFFSET = 1
+
+/**
+ *
+ */
+const TreeNode = ({
+  styleProps,
+  value,
+  item,
+  heightChangeCallback,
+}) => {
+  const [bind, { height: viewHeight }] = useMeasure()
+  const viewHeightRef = useRef(viewHeight)
+  const wrapperRef = useRef()
+
+  useEffect(() => {
+    heightChangeCallback({ height: viewHeight, key: item.key })
+    viewHeightRef.current = viewHeight
+  }, [viewHeight])
+
+  const isFirstColumn = item.viewInfo.table.column === 1
+
+  /* ### Height Animation ###
+    - viewHeight is 0 at mount and is later populated by the height observer hook.
+    - prevent height animation when second column first item is leaving because
+      y transform animation and overflow hidden with height animation looks weird and janky.
+  */
+
+  const width = isFirstColumn ? `${TABLE_FIRST_COLUMN_WIDTH}rem` : '70%'
+
+  useEffect(() => {
+    // mutating width using ref because changing width using inline style breaks resize observer :(
+    if (wrapperRef.current) wrapperRef.current.style.width = width
+  }, [width])
+
+  useEffect(() => () => heightChangeCallback({ height: null, key: item.key }), [])
```

Comment:
Is there a reason this unmount handler is in a separate `useEffect` rather than in the one that calls `heightChangeCallback` on L50?